MapleServer2/Types/SkillTab.cs

```diff
@@ -1,74 +1,54 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
-using System.Runtime.InteropServices;
-using MaplePacketLib2.Tools;
 
-namespace Maple2.Data.Types {
-    public class SkillTab : IByteSerializable {
+namespace MapleServer2.Types {
+    public class SkillTab {
         public long Id { get; set; }
         public string Name { get; private set; }
-        public ICollection<SkillTabEntry> Skills => skills;
 
-        private List<SkillTabEntry> skills;
+        private Dictionary<int, Skill> Skills;
 
-        public SkillTab(string name, ICollection<SkillTabEntry> entries = null) {
+        public SkillTab(string name, Dictionary<int, Skill> skills = null) {
+            this.Id = 0x000032DF995949B9; // temporary hard coded id
             this.Name = name;
-            this.skills = entries?.ToList() ?? new List<SkillTabEntry>();
+            this.Skills = (skills != null && skills.Any()) ? skills : new Dictionary<int, Skill>();
+
+            // Add default skills
+            AddOrUpdate(Skill.skill(20000001, 1, 1)); // Swift Swimming
+            AddOrUpdate(Skill.skill(20000011, 1, 1)); // Wall Climbing
         }
 
-        public void AddOrUpdate(in SkillTabEntry skill) {
-            // Iterate backwards for safe modification
-            for (int i = skills.Count - 1; i >= 0; i--) {
-                if (skills[i].SkillId == skill.SkillId) {
-                    skills.RemoveAt(i);
-                }
-            }
+        public void AddOrUpdate(Skill skill) {
+            // Add or update skill
+            if (Skills.ContainsKey(skill.Id)) Skills[skill.Id] = skill;
+            else Skills.Add(skill.Id, skill);
```

Comment:
While to be fair, we haven't decided on code standards yet, having everything on a single line can make the code a bit harder to read at a quick glance, which is important when you're looking for defects.

Reply:
Agreed that formatting is relative, I prefer one line ifs and elses without brackets, but if it is decided that all ifs and elses should have brackets it can be changed.